test/integration/users.test.js

```diff
@@ -121,24 +122,44 @@ describe("Users", function () {
         });
     });
 
-    it("Should update the user roles", function (done) {
+    it("Should allow updating user role when in_discord is not present and not is developer", function (done) {
+      addUser(newUser).then((newUserId) => {
+        const nonSuperUserJwt = authService.generateAuthToken({ userId: newUserId });
+        chai
+          .request(app)
+          .patch(`/users/self`)
+          .set("cookie", `${cookieName}=${nonSuperUserJwt}`)
+          .send({
+            roles: {
+              maven: true,
+            },
+          })
+          .end((err, res) => {
+            if (err) {
+              return done(err);
+            }
+
+            expect(res).to.have.status(204);
+            return done();
+          });
+      });
+    });
+    it("Should not update the user roles", function (done) {
```

Comment:
"Should not update the user roles" but when? what are the condition?
also cookie name `jwt` doesn't give context about for which the token is made for whether it's `superUserToken`, `memberToken` or `normalUserToken`

Reply:
Updated and `jwt` is basically a normal RDS user